### ISSUE TITLE
aarch64-darwin bootstrap-tools libc++.1.0.dylib LC_RPATH is wrong

### ISSUE THREAD

**Issue by ghost** [NixOS/nixpkgs#294518]
### Describe the bug

noticed in https://github.com/NixOS/nixpkgs/pull/292043#issuecomment-1986740316

filtered output of `otool -l libc++.1.0.dylib` from http://tarballs.nixos.org/stdenv-darwin/aarch64/20acd4c4f14040485f40e55c0a76c186aa8ca4f3/bootstrap-tools.cpio.bz2
```console
Load command 12
          cmd LC_LOAD_DYLIB
      cmdsize 56
         name @rpath/lib/libc++abi.dylib (offset 24)
   time stamp 2 Wed Dec 31 16:00:02 1969
      current version 1.0.0
Load command 14
          cmd LC_RPATH
      cmdsize 32
         path @loader_path/../lib (offset 12)
```

adding the name + path from the above gives us... `@loader_path/../lib/lib/libc++abi.dylib`. there is an extra lib and thus generates an invalid path and makes the lookup of libc++abi.dylib fail.

fix is to generate new bootstrap-tools with the correct paths.

**Comment by a-n-n-a-l-e-e** [comment]
this looks like it was fixed in the generation tools with PR https://github.com/NixOS/nixpkgs/pull/124228 but it seems like the aarch64 bootstrap-tools were never updated to match.

**Comment by annaleaves** [comment]
closed by #302387
